pkg/proxy/proxier.go

```diff
@@ -184,6 +192,10 @@ func (proxier *Proxier) stopProxy(service ServicePortName, info *serviceInfo) er
 
 // This assumes proxier.mu is locked.
 func (proxier *Proxier) stopProxyInternal(service ServicePortName, info *serviceInfo) error {
+	defer func() {
```

Comment:
At the time I was thinking along the lines of a "finally" clause. Probably
doesn't matter here though. I'll make it linear.

On Mon, Jun 1, 2015 at 4:43 PM, Tim Hockin <notifications@github.com> wrote:

> In pkg/proxy/proxier.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8417#discussion_r31468039>
> :
>
> > @@ -184,6 +192,10 @@ func (proxier *Proxier) stopProxy(service ServicePortName, info *serviceInfo) er
> >
> >  // This assumes proxier.mu is locked.
> >  func (proxier *Proxier) stopProxyInternal(service ServicePortName, info *serviceInfo) error {
> > +	defer func() {
>
> Why do this by defer rather than just linear code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8417/files#r31468039>
> .
>